packages/blade/src/components/Tabs/_decisions/decisions.md

```diff
@@ -0,0 +1,299 @@
+# Tabs API Decisions <!-- omit in toc -->
+
+A tab is a navigation component used in the interface to switch between different views in the same context. Tabs are contextual to the section or the page and are triggered by user interaction.
+
+<img src="./tabs-thumbnail.png" width="380" alt="Tabs thumbnail" />
+
+- [Design](#design)
+- [Features](#features)
+- [Anatomy](#anatomy)
+- [`Tabs` API](#tabs-api)
+- [`TabsItem` API](#tabsitem-api)
+- [`TabsPanel` API](#tabspanel-api)
+  - [Examples:](#examples)
+- [Basic](#basic)
+  - [Controlled, Uncontrolled](#controlled-uncontrolled)
+- [Filled Variant](#filled-variant)
+- [With Tooltip](#with-tooltip)
+- [Motion](#motion)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+
+## Design
+
+[Figma Link](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?node-id=3733%3A8981&mode=dev) to all variants of the Tabs component
+
+## Features
+
+- Controlled and Uncontrolled
+- Filled and Bordered variants
+- Horizontal and Vertical orientation
+- Medium and Large size
+
+## Anatomy
+
+<img src="./tabs-anatomy.png" width="100%" alt="Tabs thumbnail" />
+
+```jsx
```

Comment:
How about
```jsx
<Tabs>
  <TabList>
     <TabItem>
     </TabItem>
  </TabList>

  <TabPanel>
  </TabPanel>
</Tabs>

```

Reply:
TabPanel still seems very decoupled from TabItem, maybe we can nest them together somehow?

Reply:
How about something straightforward? What is the downside of this?
```jsx
<Tabs>
     <Tab value={} leading={} trailing={} title={}>
        {...the jsx content for the content of the tab}
     </Tab>
</Tabs>
```

Reply:
@chaitanyadeorukhkar while the API that you proposed looks much compact and straightforward there are major downsides to this API

### 1. Quite tricky/hack to implement

With the API that you proposed it will be quite tricky to map the JSX to the HTML structure that a typical tab requires. 

A tab's HTML structure is as follows

```html
<div>
  <div role="tablist">
    <button role="tab">tab item 1</button>
    <button role="tab">tab item 2</button>
    <button role="tab">tab item 3</button>
  </div>
  
  <div>
    <section role="tabpanel" />
    <section role="tabpanel" />
    <section role="tabpanel" />
  </div>
</div>
```

We need to achieve the above HTML structure, and to do this with the API that you proposed will get very hacky since now we need to use React.cloneElement and save those JSX nodes in state and render them in appropriate place. 

```html
<Tabs>
      // this will be the button with role="tab", need to React.cloneElement and move it to the right place, and while placing also need to make sure it has a parent of role="tablist"
     <Tab value={} leading={} trailing={} title={}>  
        // this is role="tabpanel" need to React.cloneElement and move it to the right place
        {...the jsx content for the content of the tab} 
     </Tab>
</Tabs>
```

### 2. The JSX Doesn't map well with the visual hierarchy or the HTML structure

If you notice the HTML structure pointed out in point #1 you'll see that it's 1:1 with the JSX API that i'm proposing. 

The current API that is proposed also maps 1:1 with the visual hierarchy of the component, but the second one doesn't map very well.  

![Frame 1 (11)](https://github.com/razorpay/blade/assets/35374649/9058d53f-aa5b-4a45-9111-5bfbdafa8f27)

### 3. Extensibility

The current API is also much more easier to extend in case any future usecase arises, for example let's say we want to use these tabs like a segmented control (which works like radio buttons but doesn't associate with any "tabpanels" we can simply remove the tabpanels and change the underlying html structure to acomodate for any future usecases. 

Check how reshaped is allowing this [extensibility](https://reshaped.so/content/docs/components/tabs#using-in-forms) - if you pass `name` prop it will convert the tabs to work like a Radio. 

### 4. API Familiarity 

The current API that is proposed is also much more familiar and is similar to how many other design systems are approaching compared to the second one: 

- [Reshaped](https://reshaped.so/content/docs/components/tabs)
- [Chakra](https://chakra-ui.com/docs/components/tabs/usage)
- [MUI](https://mui.com/material-ui/react-tabs/#experimental-api)
- [Atlassian](https://atlassian.design/components/tabs/examples)
- [RadixUI](https://www.radix-ui.com/primitives/docs/components/tabs)
- [HeadlessUI](https://headlessui.com/react/tabs)

Reply:
Added

Reply:
https://github.com/razorpay/blade/pull/1693#discussion_r1345552656

Also changed the naming of the current components to not use the plural suffix. instead of TabsItem -> TabItem